presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -348,14 +393,33 @@ public Type getType()
 
         public void processPage(GroupByIdBlock groupIds, Page page)
         {
-            if (step.isInputRaw()) {
+            if (!this.orderByChannels.isEmpty()) {
+                checkArgument(step == Step.SINGLE && pagesIndex.isPresent(), "ORDER BY in aggregations is supported only for single step aggregations");
+                pagesIndex.ifPresent(index -> index.addPage(page));
```

Comment:
I understand, but i think it's intellij's problem, not with the code. If we want to avoid the warning in IDE, we can introduce static `get` method that circumvents it: `get(pagesIndex).addPage(page)` or something... anyway,  I don't like an `if` in the code which is only there to satisfy IDE. Misleading -- I didn't realize it's asserted in previous line so someone else could miss that too. (My personal preference so far is to ignore the warning.)